Sources/DependencyClients/TmuxService.swift

```diff
@@ -0,0 +1,206 @@
+import Foundation
+import Models
+
+struct TmuxService: Sendable {
+  private let config: Models.SSHServerConfiguration
+  private let sshClient: SSHClient
+
+  init(config: Models.SSHServerConfiguration) {
+    self.config = config
+    self.sshClient = SSHClient()
+  }
+
+  func hasSession(_ name: String) async throws -> Bool {
+    do {
+      let connectionManager = await SSHConnectionPool.shared.manager(for: config)
+      return try await connectionManager.withConnection { connection in
+        let escapedName = escapeShellArgument(name)
+        let command = "tmux has-session -t \(escapedName) 2>/dev/null && echo 'exists' || echo 'not found'"
+        let result = try await connection.exec(command)
+        return result.trimmingCharacters(in: .whitespacesAndNewlines) == "exists"
+      }
+    } catch {
+      if error is CancellationError {
+        throw SSHError.commandFailed(
+          "Tmux session check was cancelled. This may be due to network issues or server timeout.")
+      }
+
+      throw error
+    }
+  }
+
+  func newSession(name: String, path: String) async throws {
+    do {
+      let connectionManager = await SSHConnectionPool.shared.manager(for: config)
+      try await connectionManager.withConnection { connection in
+        let escapedName = escapeShellArgument(name)
+        let escapedPath = escapeShellArgument(path)
+        let command = "tmux new-session -d -s \(escapedName) -c \(escapedPath)"
+        do {
+          _ = try await connection.exec(command)
+          await AppLogger.shared.log(
+            "Created tmux session: \(name) at \(path)",
+            level: .info,
+            category: .workspace
+          )
+        } catch {
+          if case let SSHError.commandFailed(message) = error,
+            message.lowercased().contains("duplicate session") {
+            await AppLogger.shared.log(
+              "Using existing tmux session: \(name)",
+              level: .info,
+              category: .workspace
+            )
+            return
+          }
+          throw error
+        }
+      }
+    } catch {
+      if error is CancellationError {
+        throw SSHError.commandFailed(
+          "Tmux session creation was cancelled. This may be due to network issues or server timeout."
+        )
+      }
+
+      throw error
+    }
+  }
+
+  func newOrReplaceServerWindow(name: String) async throws {
+    do {
+      let connectionManager = await SSHConnectionPool.shared.manager(for: config)
+      try await connectionManager.withConnection { connection in
+        let hasExisting = try await hasSession(name)
+        if hasExisting {
+          let escapedName = escapeShellArgument(name)
+          let killCommand = "tmux kill-session -t \(escapedName)"
+          _ = try await connection.exec(killCommand)
+        }
+
+        let workspacePath = "$HOME"
+        try await newSession(name: name, path: workspacePath)
```

Comment:
Doing it this way is inviting time-of-check time-of-use bugs.
Use the "newSession" return value as a way to distinguish between states